templates/shift-scheduling/main.go

```diff
@@ -1,370 +1,248 @@
 					}
-				}
-				return ok && len(workers) > 0
-			},
-			func() store.Store {
-				// Pop a new worker from the list.
-				worker := workers[0]
-				// Note that workers is a pointer to the variable defined above.
-				workers = workers[1:]
-
-				// First we assign the worker to the current shift.
-				changes := []store.Change{
-					shifts.AtLeast(i, worker),
-					shifts.AtMost(i, worker),
-				}
 
-				// Update happiness (i.e. are the preferences considered).
-				shiftType := typeMap[i%3]
-				if pref, ok := prefShiftType[worker]; ok {
-					if pref == shiftType {
-						changes = append(changes, happiness.Set(happiness.Get(s)+1))
+					if durationApart < 24*time.Hour {
+						lessThan10hPerDay.NewTerm(a2.Duration.Hours(), x.Get(a2))
 					}
-				} else {
-					// In case the worker has no preferences we assume they are
-					// happy.
-					changes = append(changes, happiness.Set(happiness.Get(s)+1))
-				}
-
-				s = s.Apply(changes...)
-				s = s.Propagate(constraint.propagate)
 
-				// At last we update the worker count in case some future shifts
-				// are fixed now.
-				workerMap := map[int]bool{}
-				for j := 0; j < nShifts; j++ {
-					if value, ok := shifts.Domain(s, j).Value(); ok {
-						workerMap[value] = true
+					if durationApart < 7*24*time.Hour {
+						lessThan40hPerWeek.NewTerm(a2.Duration.Hours(), x.Get(a2))
 					}
 				}
-
-				return s.Apply(workerCount.Set(len(workerMap)))
-			})
-	}).Validate(func(s store.Store) bool {
-		// The store is operationally valid if exactly one worker is assigned
-		// to a shift and no constraints are violated.
-		if !shifts.Singleton(s) {
-			return false
-		}
-
-		return !constraint.isProvenInfeasible(s)
-	}).Bound(func(s store.Store) store.Bounds {
-		// In order to make the search more efficient we define bounds.
-		// Given a partial schedule, `Bounds` returns a lower and an upper bound
-		// around the final schedule value once all workers have been assigned.
-		// As a rule of thumb: the better the bounds, the faster the search.
-
-		// The way we create new stores is that we never remove
-		// workers from a fixed shift. Thus a trivial bound is
-		// to assign a new worker to each unassigned shift.
-		// Likewise an upper bound on happiness is that each new worker
-		// is also happy.
-		wcUpper := workerCount.Get(s)
-		happinessUpper := happiness.Get(s)
-		for i := 0; i < nShifts; i++ {
-			d := shifts.Domain(s, i)
-			if _, ok := d.Value(); !ok {
-				wcUpper++
-				happinessUpper++
 			}
 		}
+	}
 
-		// For the lower bound we simply take the current worker count.
-		// As we know we need at least distinct 3 workers per day, we can
-		// take the maximum.
-		wcLower := workerCount.Get(s)
-		if wcLower < 3 {
-			wcLower = 3
-		}
+	solver, err := mip.NewSolver("highs", m)
+	if err != nil {
+		return schema.Output{}, err
+	}
+	options := mip.NewSolveOptions()
+	options.SetVerbosity(mip.Off)
+	err = options.SetMIPGapRelative(0.999)
```

Comment:
That gap could be a `const` variable somewhere at the top of the file

Reply:
sure